testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksCheckpointsCompatibility.java

```diff
@@ -378,6 +379,91 @@ private void testCheckpointNullStatisticsForRowType(Consumer<String> sqlExecutor
         }
     }
 
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, PROFILE_SPECIFIC_TESTS})
+    public void testTrinoWriteStatsAsJsonDisabled()
+    {
+        String tableName = "test_dl_checkpoints_write_stats_as_json_disabled_trino_" + randomTableSuffix();
+        testWriteStatsAsJsonDisabled(sql -> onTrino().executeQuery(sql), tableName, "delta.default." + tableName);
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, PROFILE_SPECIFIC_TESTS})
+    public void testDatabricksWriteStatsAsJsonDisabled()
+    {
+        String tableName = "test_dl_checkpoints_write_stats_as_json_disabled_databricks_" + randomTableSuffix();
+        testWriteStatsAsJsonDisabled(sql -> onDelta().executeQuery(sql), tableName, "default." + tableName);
+    }
+
+    private void testWriteStatsAsJsonDisabled(Consumer<String> sqlExecutor, String tableName, String qualifiedTableName)
+    {
+        onDelta().executeQuery(format(
+                "CREATE TABLE default.%s" +
+                        "(a_number INT, a_string STRING) " +
+                        "USING DELTA " +
+                        "PARTITIONED BY (a_number) " +
+                        "LOCATION 's3://%s/databricks-compatibility-test-%1$s' " +
+                        "TBLPROPERTIES (" +
+                        " delta.checkpointInterval = 5, " +
+                        " delta.minWriterVersion = 3, " +
+                        " delta.checkpoint.writeStatsAsJson = false)",
+                tableName, bucketName));
+
+        try {
+            sqlExecutor.accept("INSERT INTO " + qualifiedTableName + " VALUES (1,'ala')");
+
+            assertThat(onTrino().executeQuery("SHOW STATS FOR delta.default." + tableName))
```

Comment:
Out of curiosity: Introducing this feature means that we'll have up-to-date stats while writing via Trino the Delta Lake tables ?

Reply:
Sorry, I'm not sure if I understand your question correctly. This change is similar to kill switch. We were always writing `stats` and `stats_parsed` previously. Now, we don't write those fields based on `delta.checkpoint.writeStatsAsJson` and `delta.checkpoint.writeStatsAsStruct`.